.evergreen/run-async-std-tests.sh

```diff
@@ -3,6 +3,13 @@
 set -o errexit
 
 . ~/.cargo/env
-RUST_BACKTRACE=1 cargo test --no-default-features --features async-std-runtime
-RUST_BACKTRACE=1 cargo test sync --no-default-features --features sync
-RUST_BACKTRACE=1 cargo test --doc sync --no-default-features --features sync
+
+if [ "$SINGLE_THREAD" = true ]; then
+	OPTIONS="-- --test-threads=1"
```

Comment:
we might also want to set `-j 1` here in case the test runner is still running them concurrently even with a single thread. Maybe that will help squash that last set of failures we're seeing.

Reply:
The `j` flag doesn't affect anything besides building. From the `--help`:

```
The `--jobs` argument affects the building of the test executable but does
not affect how many jobs are used when running the tests. The default value
for the `--jobs` argument is the number of CPUs. If you want to control the
number of simultaneous running test cases, pass the `--test-threads` option
to the test binaries:

    cargo test -- --test-threads=1
```